usr/sbin/regionsrv-enabler-azure

```diff
@@ -96,21 +126,34 @@ def run_command(command):
 utils.start_logging()
 service_name = 'guestregister'
 license_type = get_license_type()
-
-if ('BYOS' in license_type and has_license_changed(license_type) and
-    not utils.uses_rmt_as_scc_proxy()):
-    run_command(['registercloudguest', '--clean'])
-    run_command(['systemctl', 'disable', service_name])
-    update_license_cache(license_type)
+update_server = utils.get_smt()
+
+if ('BYOS' in license_type and not utils.uses_rmt_as_scc_proxy() and
+    not utils.is_scc_connected()
+):
+    # system license type has been set to BYOS via AHB but
+    # neither has been registered to the update infrastructure with a reg code
+    # nor to SCC
+    license_has_changed = has_license_changed(license_type)
+    if (license_has_changed or
+        (update_server and utils.is_registered(update_server.get_FQDN()))
+    ):
+        # either license has changed or
+        # system was previously registered to update infrastructure
+        if license_has_changed:
```

Comment:
This condition is tested twice in consecutive conditionals, on line 138 and then again here. Can be changed to (starting at line 137)

```
if has_license_changed(license_type):
    update_license_cache(license_type)
    if update_server and utils.is_registered(update_server.get_FQDN()):
        run_command(['registercloudguest', '--clean'])
        run_command(['systemctl', 'disable', service_name])
```

to avoid checking the same condition in consecutive instructions and make the flow a little easier to follow.

Reply:
We should not do that, the de-registration must happen even if license hasnt changed
I will think a better way to check the conditions

Reply:
I think you meant outside the if
```python
if has_license_changed(license_type):
    update_license_cache(license_type)
if update_server and utils.is_registered(update_server.get_FQDN()):
    run_command(['registercloudguest', '--clean'])
    run_command(['systemctl', 'disable', service_name])
```